tools/OpenTelemetryYoutube/OpenTelemetryYoutube/Program.cs

```diff
@@ -0,0 +1,133 @@
+
+            var youtubeService = new YouTubeService(new BaseClientService.Initializer
+            {
+                HttpClientInitializer = credential,
+                ApplicationName = this.GetType().ToString(),
+            });
+
+            var channelsListRequest = youtubeService.Channels.List("snippet,contentDetails,statistics");
+            channelsListRequest.Mine = true;
+
+            // Retrieve the contentDetails part of the channel resource for the authenticated user's channel.
+            var channelsListResponse = await channelsListRequest.ExecuteAsync();
+
+            foreach (var channel in channelsListResponse.Items)
+            {
+                await ProcessChannel(youtubeService, channel);
+            }
+        }
+
+        private static async Task ProcessChannel(YouTubeService youtubeService, Channel channel)
+        {
+            // From the API response, extract the playlist ID that identifies the list
+            // of videos uploaded to the authenticated user's channel.
+            var uploadsListId = channel.ContentDetails.RelatedPlaylists.Uploads;
+
+            Console.WriteLine("Videos in list {0}", uploadsListId);
+
+            var nextPageToken = "";
+            while (nextPageToken != null)
+            {
+                var playlistItemsListRequest = youtubeService.PlaylistItems.List("snippet,contentDetails,status");
+                playlistItemsListRequest.PlaylistId = uploadsListId;
+                playlistItemsListRequest.MaxResults = 50;
+                playlistItemsListRequest.PageToken = nextPageToken;
+
+                // Retrieve the list of videos uploaded to the authenticated user's channel.
+                var playlistItemsListResponse = await playlistItemsListRequest.ExecuteAsync();
+
+                foreach (var playlistItem in playlistItemsListResponse.Items.Where(q => q.Status.PrivacyStatus == "private"))
+                {
+                    // Print information about each video.
+                    Console.WriteLine("{0} ({1})", playlistItem.Snippet.Title, playlistItem.Snippet.ResourceId.VideoId);
+
+                    var videoItemRequest = youtubeService.Videos.List("snippet,contentDetails,statistics");
+                    videoItemRequest.Id = playlistItem.Snippet.ResourceId.VideoId;
+
+                    var videoItemResponse = await videoItemRequest.ExecuteAsync();
+
+                    foreach (var videoItem in videoItemResponse.Items)
+                    {
+                        await ProcessVideo(youtubeService, videoItem);
+                    }
+                }
+
+                nextPageToken = playlistItemsListResponse.NextPageToken;
+            }
+        }
+
+        private static async Task ProcessVideo(YouTubeService youtubeService, Video videoItem)
+        {
+            var time = XmlConvert.ToTimeSpan(videoItem.ContentDetails.Duration);
+
+            // video that doesn't start with 202x and has totalMinutes > 30, we will process
```

Comment:
```suggestion
            // video that doesn't start with 202 and has totalMinutes > 30, we will process
```

What does 202 means?

Reply:
2020 (the year), so when we move to 2021....and keep going.

Reply:
Oh, this is the year =). Can you please add a comment

Reply:
just added. can you re-check?